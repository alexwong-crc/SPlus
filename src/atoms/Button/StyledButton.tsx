import styled from "styled-components";

interface IStyledButton {
  type: "button" | "submit" | "reset";
  showText?: boolean;
}

const StyledButton = styled.button<IStyledButton>`
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  background-color: ${({ theme }): string => theme.action.main.code};
  color: ${({ theme }): string => theme.action.main.text};
  border: none;
  font-size: 0.8rem;
  position: relative;
  :active {
    background-color: ${({ theme }): string => theme.action.dark.code};
  }
  :hover {
    cursor: pointer;
  }
  :focus {
    outline: none;
  }
  :disabled {
    cursor: default;
    opacity: 0.6;
    background-color: ${({ theme }): string => theme.action.main.code};
  }
  span.text {
    opacity: ${({ showText }): number => (showText ? 1 : 0)};
  }
  span.icon {
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
  }
`;

export default StyledButton;
