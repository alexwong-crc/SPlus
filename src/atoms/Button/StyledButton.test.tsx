import { shallow } from "enzyme";
import React from "react";
import StyledButton from "./StyledButton";
import defaultProps from "../../../services/tests/defaultProps";

describe("StyledButton Component", () => {
  it("should snapshot the styled button", () => {
    const component = shallow(<StyledButton type="button" {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
