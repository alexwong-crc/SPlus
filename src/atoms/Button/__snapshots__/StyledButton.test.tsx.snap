// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component should snapshot the styled button 1`] = `
.c0 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  background-color: #76C56F;
  color: #FFFFFF;
  border: none;
  font-size: 0.8rem;
  position: relative;
}

.c0:active {
  background-color: #32AA27;
}

.c0:hover {
  cursor: pointer;
}

.c0:focus {
  outline: none;
}

.c0:disabled {
  cursor: default;
  opacity: 0.6;
  background-color: #76C56F;
}

.c0 span.text {
  opacity: 0;
}

.c0 span.icon {
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<button
  className="c0"
  type="button"
/>
`;
